cdc/log.cc

```diff
@@ -1296,6 +1300,23 @@ struct process_change_visitor {
             auto log_ck = _builder.allocate_new_log_row(end_operation);
             _builder.set_clustering_columns(log_ck, rt.end);
         }
+
+        // #6900 - remove stored row data (for postimage)
+        // if it falls inside the clustering range of the
+        // tombstone.
+        bound_view::compare cmp(_builder.base_schema());
+        auto sb = rt.start_bound();
+        auto eb = rt.end_bound();
+        auto i = _clustering_row_states.begin();
+        auto e = _clustering_row_states.end();
+        while (i != e) {
+            auto& ck = i->first;
+            if (cmp(sb, ck) && !cmp(eb, ck)) {
+                i = _clustering_row_states.erase(i);
+                continue;
```

Comment:
wouldn't _clustering_row_states.erase_if be of use here?

Reply:
There is no `erase_if`member function. There is `std::erase_if(std::unordered_map, pred)`in c++20 though. Sorry, I am so old school when it comes to maps.

Reply:
Generally we should avoid "raw loops"

Reply:
@elcallio there is `erase_if` for vectors too
https://en.cppreference.com/w/cpp/container/vector/erase2

and there is `remove_if` for iterators

Reply:
Yeah, I know. But I am old enough to feel weird about std routines that don't operate on iterators only.

Reply:
fixed